docs/rfc/rfc-010-p2p-light-client.rst

```diff
@@ -0,0 +1,127 @@
+==================================
+RFC 010: Peer to Peer Light Client
+==================================
+
+Changelog
+---------
+
+- 2022-01-21: Initial draft (@tychoish)
+
+Abstract
+--------
+
+The dependency on access to the RPC system makes running or using the light
+client more complicated than it should be, because it is difficult to get
+access to the RPC endpoints on a running node. There is no deep dependency for
+the light client on the RPC system, and there is a persistent notion that
+"make a p2p light client" is a solution to this operational limitation. This
+document explores the implications and requirements of implementing a
+p2p-based light client, as well as the possibilities afforded by this
+implementation.
+
+Background
+----------
+
+High Level Design
+~~~~~~~~~~~~~~~~~
+
+From a high level, the light client P2P implementation, is relatively straight
+forward, and the P2P-backed statesync implementation that took place during
+the 0.35 cycle. The light client only really needs to be able to request (and
+receive) a `LightBlock` at a given height. To support this, a new Reactor
+would run on every full node and validator which would be able to service
+these requests. The workload would be entirely request-response, and the
+implementation of the reactor would likely be very straight forward, and the
+implementation of the provider is similarly relatively simple. 
+
+The complexity of the project focuses around peer discovery, handling when
```

Comment:
Would the peer footprint of a light client be (substantially) different from the peer footprint of the node _qua_ node?

Reply:
Yes, by "peer footprint" I meant mainly the number and identity of peers, and the amount of data needed to communicate with them. It seems as though a p2p light client would not want to talk to a radically different set/volume of peers than a regular node.

Reply:
I presume that most of a light client would want to talk to 3 nodes at a time, potentially more here and there as nodes go off line and return, but in general, should be pretty small. 

I worry more about the impact of light clients on full nodes.